resolwe_bio/processes/reads_processing/bqsr.py

```diff
@@ -0,0 +1,153 @@
+        reference = DataField('genome:fasta', label='Reference genome file', required=True)
+        known_sites = ListField(
+            DataField('variants:vcf', description='Known sites in (.vcf).'),
+            label='One or more databases of known polymorphic sites used to exclude regions around known '
+                  'polymorphisms from analysis.',
+            required=True
+        )
+        intervals = DataField(
+            data_type='bed',
+            label='One or more genomic intervals over which to operate. This field is opotional, but it can speed up '
+                  'the process by restricting calculations to specific genome regions.',
+            required=True
+        )
+        read_group = StringField(
+            label='Replace read groups in a BAM file.This argument enables the user to replace all read groups in the '
+                  'INPUT file with a single new read group and assign all reads to this read group in the OUTPUT BAM '
+                  'file. Addition or replacement is performed using Picard\'s AddOrReplaceReadGroups tool. Input '
+                  'should take the form of -<name><space><value> delimited by a tab, e.g. "-ID 1\t-PL Illumina\t-SM '
+                  'sample_1". See https://software.broadinstitute.org/gatk/documentation/tooldocs/4.1.2.0'
+                  '/picard_sam_AddOrReplaceReadGroups.php for more information on tag names. Note that PL, LB, '
+                  'PU and SM are require fields.',
+            required=False,
+            default=''
+        )
+
+    class Output:
+        """Output fields to BaseQualityScoreRecalibrator."""
+        bam = FileField(label='Base quality score recalibrated BAM file')
+        bai = FileField(label='Index of base quality score recalibrated BAM file')
+        stats = FileField(label='Alignment statistics')
+        bigwig = FileField(label='BigWig file')
+        species = StringField(label='Species')
+        build = StringField(label='Build')
+        recal_table = FileField(label='Recalibration tabled')
+
+    def run(self, inputs, outputs):
+        # Prepare output file names.
+        bam = os.path.basename(inputs.bam.bam.path)
+        file_name = bam[:-4]
+        bam_rg = f'{file_name}_RG.bam'
+
+        species = inputs.bam.species
+
+        # Parse read_group argument from a string, delimited by a \t and space
+        # into a form that will be accepted by AddOrReplaceReadGroups tool.
+        # E.g. '-LB DAB\t-PL Illumina\t-PU barcode -SM sample1' should become
+        # ['-LB', 'DAB', '-PL', 'Illumina', '-PU', 'barcode', '-SM', 'sample1']
+        # prepended by INPUT and OUTPUT.
+        if len(inputs.read_group) > 0:
+            arrg = [
+                '--INPUT', f'{inputs.bam.bam.path}',
+                '--OUTPUT', f'{bam_rg}'
+            ]
+
+            for x in inputs.read_group.split('\t'):
+                arrg.extend(x.split(' '))
```

Comment:
Should this be split up at only the first space encountered?
e.g. is this valid, `'-LB DAB 123\t-PL Illumina 123'` and should it parse to `['-LB', 'DAB 123', '-PL', 'Illumina 123']`?

Reply:
The string should split only on the first space. Your suggestion doesn't parse well with my current solution.

```
x = '-LB DAB 123\t-PL Illumina 123'
x.split(' ')
['-LB', 'DAB', '123\t-PL', 'Illumina', '123']
```
Do you happen to have something more robust on your mind or should I think some more about this?

Reply:
I don't know.
One option is basic:text:, to have newlines as separator. And something less common than `<space>` as delimiter. E.g. `=`

(split has a `max` option)

Reply:
I think having a `=` instead of space solves the issue with spaces in the name, which we should respect because it's the only [non-printing character](https://software.broadinstitute.org/gatk/documentation/tooldocs/4.1.2.0/picard_sam_AddOrReplaceReadGroups.php) allowed in the name. 

`basic:text` is a nice idea, but I don't see a whole lot of benefits, other than having the explicit new line. My guess would be that initially it would probably be a bit less intuitive. I would leave the `\t` as a separator for now. I have thought of adding something else, like a pipe (`|`), but some people may have problems finding that on their keyboards.

Reply:
@romunov What about handling read info information [this](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/variant_calling/vc_chemut.yml#L53) way? You can define the type (and value range) of each input, and in the process control the delimiters. Its more inputs, but in the end it will be more robust.

Reply:
If we go down that road, we would have to implement (and test) 13 tags whereas the current implementation is agnostic to the number of tags used. Perhaps we can do this for version 2.0?

Reply:
It was just a suggestion, as I remembered that we were handling this information using separate fields in the past.

Reply:
A valid suggestion nonetheless. Another case against this type of input would be the long list of parameters, which would be a PITA to fill out if you want to repeat this over and over. Until we get the auto-prefilled-parameters-based-on-previous-run feature, this would be somewhat cumbersome.